webpack/scenes/ContentViews/Table/ContentViewsTable.js

```diff
@@ -41,155 +44,90 @@ const ContentViewTable = () => {
-      next_version: rowInfo.cvNextVersion,
-    });
+  const openPublishModal = (cvInfo) => {
+    setActionableCvDetails(cvInfo);
     setIsPublishModalOpen(true);
   };
 
-  const openPromoteModal = (rowInfo) => {
+  const openPromoteModal = (cvInfo) => {
     dispatch(getEnvironmentPaths());
-    setActionableCvDetails({
-      id: rowInfo.cvId.toString(),
-      latestVersionId: rowInfo.latestVersionId,
-      latestVersionEnvironments: rowInfo.latestVersionEnvironments,
-      latestVersionName: rowInfo.latestVersionName,
-    });
+    setActionableCvDetails(cvInfo);
     setIsPromoteModalOpen(true);
   };
 
-  const openDeleteModal = (rowInfo) => {
-    dispatch(getContentViewDetails(rowInfo.cvId));
-    dispatch(getContentViewVersions(rowInfo.cvId));
-    setActionableCvDetails({
-      id: rowInfo.cvId.toString(),
-      name: rowInfo.cvName,
-      environments: rowInfo.environments,
-      versions: rowInfo.versions,
-    });
+  const openDeleteModal = (cvInfo) => {
+    dispatch(getContentViewDetails(cvInfo.id));
+    dispatch(getContentViewVersions(cvInfo.id));
+    setActionableCvDetails(cvInfo);
     setIsDeleteModalOpen(true);
   };
 
-  useDeepCompareEffect(
-    () => {
-      // Prevents flash of "No Content" before rows are loaded
-      const tableStatus = () => {
-        if (typeof cvResults === 'undefined' || status === STATUS.ERROR) return status; // will handle errored state
-        const resultsIds = Array.from(cvResults.map(result => result.id));
-        // All results are accounted for in row mapping, the page is ready to load
-        if (resultsIds.length === rowMappingIds.length &&
-          resultsIds.every(id => rowMappingIds.includes(id))) {
-          return status;
-        }
-        return STATUS.PENDING; // Fallback to pending
-      };
-
-      const { results } = response;
-      if (status === STATUS.ERROR) {
-        setCvTableStatus(tableStatus());
-      }
-      if (!loadingResponse && results) {
-        setCvResults(results);
-        setCurrentStep(1);
-        const { newRowMappingIds, ...tableData } = tableDataGenerator(results);
-        setTable(tableData);
-        setRowMappingIds(newRowMappingIds);
-        setCvTableStatus(tableStatus());
-      }
-      return () => {
-        // This sets the loading state so that the table doesn't flicker on return
-        setCvTableStatus(STATUS.PENDING);
-      };
-    },
-    [response, status, loadingResponse, setTable, setRowMappingIds,
-      setCvResults, setCvTableStatus, setCurrentStep, cvResults, rowMappingIds],
-  );
-
-  const onCollapse = (_event, rowId, isOpen) => {
-    let rows;
-    if (rowId === -1) {
-      rows = table.rows.map(row => ({ ...row, isOpen }));
-    } else {
-      rows = [...table.rows];
-      rows[rowId].isOpen = isOpen;
-    }
+  const actionsWithPermissions = (cvInfo) => {
+    const { version_count: cvVersionCount, generated_for: generatedFor, permissions } = cvInfo;
```

Comment:
@parthaa I had to fix this during a rebase. I hope `generated_for` is the correct key name here..